plugin/evm/vm.go

```diff
@@ -642,7 +670,7 @@ func (vm *VM) buildBlock(context.Context) (snowman.Block, error) {
 	// We call verify without writes here to avoid generating a reference
 	// to the blk state root in the triedb when we are going to call verify
 	// again from the consensus engine with writes enabled.
-	if err := blk.verify(false /*=writes*/); err != nil {
+	if err := blk.verify(proposerVMBlockCtx, false /*=writes*/); err != nil {
```

Comment:
AFAIK in case this block activates a precompile with a predicate, they won't be checked since `currentRules` in vm will not activate the precompile for this block (as it won't accepted yet). This is fine as we also do the same thing for precompiles that would change the validity of txs/the block by precompiles activated at the same block. E.g fee config manager and coinbase would depend on the state of the previous block rather than the generated one. But we should clearly document this expected behaviour with both comments and the function signatures. 

cc @aaronbuchwald

Reply:
I added a comment explaining this to `Block.verify`, but didn't alter any function signatures. Do you have any suggestions on how best to do so to document this? Where is the precompile activation check done?

Reply:
Looping back here, do we need to add a special case to make sure that no warp messages should be allowed to be received in the activation block of the warp precompile? Doesn't need to be in this PR, but it is something we should track if so.

Reply:
Yes this is a bug as implemented and will not be merged in its current form.

This should use the rule set for the block instead of `vm.currentRules()`.